sast: added scan stats to coverity

jperezdealgaba opened the pull request:
Solves: https://issues.redhat.com/browse/OSH-769

Added stats to SARIF result file by using the coverity list command.

# Before you complete this pull request ...

Look for any open pull requests in the repository with the title "e2e-tests update" and 
see if there are recent e2e-tests updates that will be applicable to your change.

jperezdealgaba commented:
@kdudka I know it is a bit messy, but would would you mind checking the changes to the Coverity task to add the stats?
The changes start here: https://github.com/jperezdealgaba/build-definitions/blob/dd414a876d5e2aa7c28bc5789cbb29abf65baeb6/task/sast-coverity-check/0.1/sast-coverity-check.yaml#L233
and end in https://github.com/jperezdealgaba/build-definitions/blob/dd414a876d5e2aa7c28bc5789cbb29abf65baeb6/task/sast-coverity-check/0.1/sast-coverity-check.yaml#L260

When review is finished, I will add this to the original `coverity-buildless` task MR.
If you want to download the results to see them, you can try:

`oras pull quay.io/redhat-user-workloads/jperezde-tenant/test-coverity/osh-cli-container-konflux-test-2@sha256:00b108e99e3e6edb9fac38dc002cd6a318e2ff07504d0ecd33c794bc82b7911f`

Here is a working pipeline: https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/jperezde/applications/test-coverity/pipelineruns/osh-cli-container-konflux-test-2-on-pull-request-lrjjc

![Screenshot 2024-11-26 at 14 05 17](https://github.com/user-attachments/assets/ab31bd5f-5327-4c4b-aaaa-9fd5a32ec0ba)

kdudka commented:
Please do **not** update https://github.com/konflux-ci/build-definitions/pull/1411 - it needs to be merged ASAP.  This is a backward-compatible change, which can be submitted as a separate pull request and merged later.

I would rename `cov-analyzed-loc` to `cov-scanned-lines`.  `csmock-plugin-coverity` uses `cov-lines-processed` for this but I do not think the field is consumed by any automation.  So let's make the new interface consistent.

jperezdealgaba commented:
@kdudka ACK. I would not update https://github.com/konflux-ci/build-definitions/pull/1411 and will submit this MR once #1411 is merged.

I have also renamed the field to `cov-scanned-lines`.

jperezdealgaba committed:
sast: initial task for Coverity buildless

Solves: https://issues.redhat.com/browse/OSH-740

Initial version of the Coverity Buildless task. In introduces two different tasks: A task checking the availability of Coverity license and authentication token, and a task for scanning the code. The code will be scanned using coverity buildless mode, then the results are processing using csgrep and the results are later filtered using csfilter-kfp.
